### ISSUE TITLE
Alignment trouble

### ISSUE THREAD

**Issue by MfMartel** [volcic/motom-toolbox#2]
Hi,

I am trying to perform a new alignment, without the NDI cube with a rigid body that I have created on 3D architect, and I have some errors at the very beginning:
```
Error using optotrak_set_up_system (line 29)
The config file doesn't exist or it can't be read. Have you got permissions?
```

When looking it up, it seems that it can not find the correct config file. And I don't know what to put in there... In `optotrak_align_with_single_camera.m` your config file is **alignment_settings**. But alignment_settings is not in the folder, so I'm not exactly sure what's in there. The only config file in the alignment folder is "this_works_with_the_NDI_cube", so I modified this one to my setup and tried to use it, but it fails so I guess it's not (especially since you commented that you made your alignment config files with 10 markers, and the cube is 16).

```
original_camera_file = 'standard'; %initially.
config_file = 'alignment_settings'; %I tailored this for my 10 marker setup. Change it to your own. Make sure all markers are visible and bright enough. 
```
**--> What are the specs of this file?**

```
recording_file = sprintf('recording_used_for_alignment_%s.dat', datestr(now, 'yyyy-mm-dd_HH-MM-SS')); %In this case, the local directory.
rigid_body_file = 'PlanFinal_3Markers_mm'; 
```
**-->This is my new rigid body file, built with 3D architect**
```
%rigid_body_file = 'asztal'; %This can be in C:\ngidital\rigid, or local. I chose local.
new_camera_file = sprintf('Aligned_%s.cam', datestr(now, 'yyyy-mm-dd_HH-MM-SS')); %Genreate the new camera file
logfile_name = sprintf('Aligned_%s.log', datestr(now, 'yyyy-mm-dd_HH-MM-SS')); %Genreate the new camera file

%% Step 0. Initialise. Declare useful variables here so I can check what I am meddling with.
optotrak_startup; %Do the reset dance. Make the system beep.
optotrak_set_up_system(original_camera_file, config_file); %Load the settings.
fail = RigidBodyAddFromFile_euler(0, 1, rigid_body_file); % Load a rigid body as well. %% Was commented before
if(fail)
    optotrak_tell_me_what_went_wrong;
    error('Couldn''t add the specified rigid body.')
end
```

A few lines below you load some parts of the config_file you used, yet I don't find all these info on any other config_file.

```
% We need to know how the system is initialised. It's in the config file,
% but let's load it here.
fail = 0;
sensors = 0;
odaus = 0;
rigid_bodies = 0;
markers = 0;
frame_rate = 0;
marker_frequency = 0;
threshold = 0;
gain = 0;
are_we_streaming = 0;
duty_cycle = 0;
voltage = 0;
collection_time = 0;
pretrigger_time = 0;
flags = 0;
```

Sorry if it's a stupid question, but could you tell me what the structure of alignment_settings is supposed to be like? Are the info listed above the only one I need in there?

thanks!

**Comment by ha5dzs** [comment]
Hi,

You are on the right track, but I don't think you need to go that nitty-gritty with it. You just need to make a new config file for the rigid body you use, and execute a convenience script that I have put together to align the coordinate systems of all sorts of set-ups. This config file should instruct the system on how many markers have you got, and how long of a recording you want to use for the alignment.

So your rigid body has 3 markers, and let's say you want to record 30 seconds of raw data for the system to work with. This is the config file that you need:
```

[OPTOTRAK System]
NumberOfMarkers = 3
FrameFrequency = 50
MarkerFrequency = 1000
DutyCycle = 0.60
StroberVoltage = 6
InterpolationFlag = TRUE
NoFireMarkers = FALSE
StaticThresholdFlag = FALSE
StaticThresholdValue = 0
DynamicThresholdPercent = 30
MinimumGain = 160
CollectionTime = 30
PreTriggerTime = 0
StreamData = 0
DynamicDutyFlag = FALSE
BufferWaveformsFlag = FALSE
Buffer3dFlag = FALSE
BufferFullRawFlag = FALSE
GetNextFrameFlag = FALSE
ExternalSyncFlag = FALSE
ExternalTriggerFlag = FALSE
PollRealDataFlag = FALSE
bConvertOnHost = TRUE
bRigidOnHost = TRUE
MarkerType = 3
WavelengthType = 0
ModelType = 1
TimeOut = 5
bIgnoreVersionCheck = FALSE
bPerformExpectedSystemCheck = FALSE
bmode = 0
RCParam1 = 0
RCParam2 = 1
3dErrorMode = 1
```
So, with this:
-The marker frequency is set to 1000 Hz,
-The frame rate is set to 50 frames per second
-The duty cycle is 60%
-The strober voltage is 6 Volts
-...and we are using the buffer, and the rest of them pretty much don't matter.

You can adapt this to your set-up, and name it `PlanFinal_3Markers_mm.ini` , you can use `optotrak_align_my_system()`, which is a unified script that can handle many cameras and does the necessary recordings, tests, camera file generation and logging.

Make sure that everything is plugged in to port 1, and no other markers are present (or, if you must, set up the strober port table accordingly). Also make sure that your rigid body is actually working (say by default, there is about 0.5mm spread of markers are permitted) If the rigid body transform becomes impossible, the alignment script will fail.

For you and your set-up, the command you need to execute will be:
```
optotrak_align_my_system('PlanFinal_3Markers_mm.ini', 'PlanFinal_3Markers_mm.rig');
```
It will do the recording, the evaluation and the alignment. If there is a problem, it tells you without crashing (hopefully :)). I have also added some extra safeguards, so you can use the extensions in the input arguments, which I found helpful when I developed code. Assuming the final alignment error is less than 0.6mm, you can use the created camera file, which will be something like:
```
Aligned_2019-02-13_11-43.cam
```
You can then use this camera file with First Principles, or with `optotrak_set_up_system()` in your code with the toolbox. I use this script every day.


Also, for checking how the system is initialised, you can use `optotrak_get_setup_info()`. It downloads all the settings, and interprets flags too.

Hope this helps!

**Comment by MfMartel** [comment]
Hi,

Thanks! So basically the config file to use is the same as the one This_works_with_the_NDI_cube except that we change the number of markers. Somehow I thought otherwise because of the part of the code with fail, odaus...

I had some transformation issues as well I think because I had manually transformed it on 3D architect, but I made a new .rig file and it works perfectly.

Thanks again for being so responsive :)
